mappings/muse.py

```diff
@@ -1,100 +1,170 @@
-from mappings.marc import MARCMapping
+from . import utils
+from .base_mapping import BaseMapping
+from model import FRBRStatus, Record, Source
+from uuid import uuid4
+from datetime import datetime, timezone
 
 DEFAULT_PUBLISHER = 'John Hopkins University Press||'
 
 
-class MUSEMapping(MARCMapping):
-    def __init__(self, source):
-        super(MUSEMapping, self).__init__(source, {})
-        self.mapping = self.createMapping()
-    
-    def createMapping(self):
-        return {
-            'identifiers': [
-                ('001', '{0}|muse'),
-                ('010', '{z}|lccn'),
-                ('020', '{a}{z}|isbn'),
-                ('022', '{a}|issn'),
-                ('035', '{a}|oclc'),
-                ('040', '{a}|muse')
-            ],
-            'authors': [
-                ('100', '{a} {b} {c} {d}|||true'),
-                ('110', '{a} {b} {c} {n} {d}|||true'),
-            ],
-            'title': [
-                ('245', '{a} {b}'),
-                ('130', '{a}')
-            ],
-            'alternative': [
-                ('240', '{a} {k}') # Uniform Title
-            ],
-            'has_version': ('250', '{a} {b}|'),
-            'publisher': [('264', '{b}||')],
-            'spatial': ('264', '{a}'),
-            'dates': [
-                ('264', '{c}|publication_date')
-            ],
-            'languages': [('008', '||{0}')],
-            'extent': ('300', '{a}{b}{c}'),
-            'table_of_contents': ('505', '{a}'),
-            'abstract': [
-                ('500', '{a}'),
-                ('504', '{a}'),
-                ('520', '{a}'),
-            ],
-            'subjects': [
-                ('600', '{a} {d} -- {v} -- {x} -- {y} -- {z}|{2}|{0}'),
-                ('610', '{a} {d} -- {v} -- {x} -- {y} -- {z}|{2}|{0}'),
-                ('611', '{a} {d} -- {v} -- {x} -- {y} -- {z}|{2}|{0}'),
-                ('630', '{a} {p} -- {v} -- {x} -- {y} -- {z}|{2}|{0}'),
-                ('650', '{a} {b} -- {v} -- {x} -- {y} -- {z}|{2}|{0}'),
-                ('651', '{a} -- {v} -- {x} -- {y} -- {z}|{2}|{0}'),
-                ('656', '{a} -- {v} -- {x} -- {y} -- {z}|{2}|{0}')
-            ],
-            'contributors': [
-                ('260', '{f}|||manufacturer'),
-                ('700', '{a} {b} {c} {d}|||{e}'),
-                ('710', '{a} {b} {c} {d}|||{e}'),
-                ('711', '{a} {e}|||{j}'),
-            ],
-            'is_part_of': ('490', '{a}|{v}|volume'),
-            'has_part': [
-                ('856', '1|{u}|muse|text/html|{{"reader": false, "download": false, "catalog": false, "embed": true}}')
-            ]
-        }
+class MUSEMapping(BaseMapping):
```

Comment:
Feel free to remove this class altogether and just have the function map_muse_record!